jdaviz/configs/default/plugins/export/export.py

```diff
@@ -190,6 +198,59 @@ def __init__(self, *args, **kwargs):
             # on the user's machine, so export support in cubeviz should be disabled
             self.serverside_enabled = False
 
+    def _is_subset_format_supported(self):
+        """
+        Check if the format of the subset to be exported is supported.
+        """
+        if self.subset.selected == '' or self.subset.selected is None:
+            return
+
+        subset = self.app.get_subsets(self.subset.selected)
+        selected = self.subset_format.selected
+
+        self.subset_format_invalid_msg = ''
+        is_supported = True
+
+        try:
+            is_spectral = self.app._is_subset_spectral(subset[0])
+        except KeyError:
+            is_spectral = False
+
+        # Disable selecting a bad subset+format combination from the API
+        if is_spectral and selected != 'ecsv':
+            is_supported = False
+        elif not is_spectral and selected == 'ecsv':
+            is_supported = False
+
+        # raise vue message
+        if not is_supported:
+            self.subset_format_invalid_msg = (f"Export of '{self.subset.selected}' "
+                                              f"in '{selected}' format is not supported.")
+
+    @observe('subset_selected')
+    def _on_subset_selected(self, event):
+        if hasattr(self, 'subset_format'):
+            self.subset_format._update_items()
+            self._is_subset_format_supported()
+
+    # Use an observer function rather than slap the decorator on
+    # _is_subset_format_supported for clarity and to follow the
+    # convention of the method above.
+    @observe('subset_format_selected')
+    def _on_subset_format_selected(self, event):
+        self._is_subset_format_supported()
+
+    @observe('viewer_items', 'dataset_items', 'subset_items',
+             'plugin_table_items', 'plugin_plot_items')
+    def _set_relevant(self, *args):
+        if self.app.config != 'deconfigged':
+            return
+        if not (len(self.viewer_items) or len(self.dataset_items) or len(self.subset_items)
+                or len(self.plugin_table_items) or len(self.plugin_plot_items)):
+            self.irrelevant_msg = 'Nothing to export'
+        else:
+            self.irrelevant_msg = ''

```

Comment:
this diff still seems to contain a lot of changes not in the original PR that got pulled over from other PRs to main that were intentionally not backported